combined_robot_hw_tests/CMakeLists.txt

```diff
@@ -9,29 +10,47 @@ find_package(catkin REQUIRED COMPONENTS
   roscpp
 )
 
-include_directories(include ${catkin_INCLUDE_DIRS})
-
+# Declare a catkin package
 catkin_package(
-  INCLUDE_DIRS include
-  LIBRARIES ${PROJECT_NAME}
-  CATKIN_DEPENDS combined_robot_hw hardware_interface roscpp
+  INCLUDE_DIRS
+    include
+  LIBRARIES
+    ${PROJECT_NAME}
+  CATKIN_DEPENDS
+    combined_robot_hw
+    hardware_interface
+    roscpp
+    # Note: controller_manager and controller_manager_tests are only used in tests
 )
 
+
+###########
+## Build ##
+###########
+
+# Specify header include paths
+include_directories(include ${catkin_INCLUDE_DIRS})
+
 add_library(${PROJECT_NAME}
   src/my_robot_hw_1.cpp
   src/my_robot_hw_2.cpp
   src/my_robot_hw_3.cpp
   src/my_robot_hw_4.cpp
 )
 add_dependencies(${PROJECT_NAME} ${catkin_EXPORTED_TARGETS})
-target_link_libraries(${PROJECT_NAME} ${catkin_LIBRARIES} ${Boost_LIBRARIES})
+target_link_libraries(${PROJECT_NAME} ${catkin_LIBRARIES})
 
 add_executable(combined_robot_hw_dummy_app src/dummy_app.cpp)
 target_link_libraries(combined_robot_hw_dummy_app ${PROJECT_NAME} ${catkin_LIBRARIES})
 
-if(CATKIN_ENABLE_TESTING)
 
+#############
+## Testing ##
+#############
+
+if(CATKIN_ENABLE_TESTING)
   find_package(rostest REQUIRED)
+
   add_rostest_gtest(combined_robot_hw_test
```

Comment:
is there a way to `add_dependencies(.... combined_robot_hw_dummy_app)`?

Reply:
Good catch, I'll add that to the tests which use it

Reply:
Added in 41f6cff8

Reply:
Neat, thanks!